src/lib/increment.js

```diff
@@ -0,0 +1,38 @@
+/**
+* Copyright 2012-2020, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+
+'use strict';
+
+module.exports = function incrementNumeric(x, delta) {
+    if(!delta) return x;
+
+    // Note 1:
+    // 0.3 != 0.1 + 0.2 == 0.30000000000000004
+    // but 0.3 == (10 * 0.1 + 10 * 0.2) / 10
+    // Attempt to use integer steps to increment
+    var scale = 1 / Math.abs(delta);
+    if(scale < 1) scale = 1;
+    var newX = (
+        scale * x +
+        scale * delta
+    ) / scale;
```

Comment:
🐎 just avoiding this extra work when it's not doing anything :)
```suggestion
    var scale = 1 / Math.abs(delta);
    var newX;
    if(scale > 1) {
        newX = (
            scale * x +
            scale * delta
        ) / scale;
    }
    else {
        newX = x + delta;
    }
```

Reply:
Good call. Addressed in https://github.com/plotly/plotly.js/pull/5114/commits/eae05f2af523d3d0873e93921203bb6996f75e1f.